src/runtime_src/core/tools/common/ReportAie.cpp

```diff
@@ -0,0 +1,324 @@
+/**
+ * Copyright (C) 2020 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+// ------ I N C L U D E   F I L E S -------------------------------------------
+// Local - Include Files
+#include "ReportAie.h"
+#include "core/common/query_requests.h"
+#include "core/common/device.h"
+#include <boost/property_tree/json_parser.hpp>
+
+#define fmtCommon(x) boost::format("    %-22s: " x "\n")
+
+namespace qr = xrt_core::query;
+
+const uint32_t major = 1;
+const uint32_t minor = 0;
+const uint32_t patch = 0;
+
+boost::property_tree::ptree
+populate_aie(const xrt_core::device * device, const std::string& desc)
+{
+  boost::property_tree::ptree pt;
+  std::string aie_data;
+  pt.put("description", desc);
+  boost::property_tree::ptree graph_array;
+  boost::property_tree::ptree _pt;
+
+  try {
+    aie_data = xrt_core::device_query<qr::aie_metadata>(device);
+    std::stringstream ss(aie_data);
+    boost::property_tree::read_json(ss, _pt);
+  } catch (const std::exception& ex){
+    pt.put("error_msg", ex.what());
+    return pt;
+  }
+
+  try {
+    if(_pt.get<uint32_t>("schema_version.major") != major ||
+         _pt.get<uint32_t>("schema_version.minor") != minor ||
+         _pt.get<uint32_t>("schema_version.patch") != patch ) {
+      pt.put("error_msg", (boost::format("major:minor:patch [%d:%d:%d] version are not matching")
+          % _pt.get<uint32_t>("schema_version.major")
+          % _pt.get<uint32_t>("schema_version.minor")
+          % _pt.get<uint32_t>("schema_version.patch")));
+      return pt;
+    }
+
+  pt.put("schema_version.major", major);
+  pt.put("schema_version.minor", minor);
+  pt.put("schema_version.patch", patch);
+
+ /*
+ * sample AIE json which can be parsed

```

Comment:
All these stars do not add a lot...
For fun, try to copy paste your comment to use it as an example. :-(

Reply:
How do you solve the copy paste issue?
Easy solution: do not put decorations in the middle of the comments!
Otherwise if you want to make the code worse, you could add some random smileys, some cuisine recipe... :-)